sql/sqlserver/server_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tobias Schottdorf
+
+package sqlserver
+
+import (
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/sql/sqlwire"
+)
+
+func TestProcessSelect(t *testing.T) {
+	s := (*Server)(nil) // enough for now
+
+	vInt := int64(5)
+	vNum := "3.14159"
+	vStr := "two furs one cub"
+	vBool := true
+
+	dInt := sqlwire.Datum{IntVal: &vInt}
+	dFloat := func() sqlwire.Datum {
```

Comment:
If you never saw the shadowing error, how do you know it would even come up? The reference to `err` below is safe - this binding of `err` is never accessed before the shadow happens.

If doing this without a closure triggers a shadowing error, it's a bug in `vet`.